### PR TITLE
audacity: overhaul and upgrade

### PR DESCRIPTION
**Pull request by RJVB**
#### Description
This upgrade and overhaul of the Audacity Portfile hopefully adresses the
building issues currently affecting it.
- port:audacity-legacy (v2.4.2) is renamed to audacity2-legacy and is the last
version to provide the old project format and (presumably) compatability with
older OS versions.
- the current port:audacity (3.0.2.129) is renamed to audacity3-legacy; it
(presumably) provides compatibility with OS X 10.7 and 10.8 .
- port audacity is upgraded to git/HEAD (3.2.5.x; development goes quickly ATM).
It provides a port:audacity-wxWidgets subport that is compulsory on 10.9 and
10.10; on newer OS versions it will also build against port:wxWidgets-3.2 .
- port:audacity-wxWidgets is a pure build dependency

Also relinquishing my maintainership.

Fixes: https://github.com/macports/macports-ports/pull/17682
Closes: https://trac.macports.org/ticket/66801

###### Type(s)

- [x ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

OS X 10.9.5

###### Verification 
Have you

- [x ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [X] tested basic functionality of all binary files?
- [X] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by RJVB** [comment]
Looking into the SDK mix-up :-/

File: audio/audacity/Portfile [hunk 1]
```diff
@@ -81,34 +81,42 @@ if {${subport} eq "${name}-legacy"} {
     # get the source tarball from github because it contains all required external libs
     # incl. those not in MacPorts.
     github.setup    audacity audacity 2.4.2 Audacity-
-    revision        2
+    revision        4
```
Comment (reneeotten) [inline audio/audacity/Portfile:84]: seems to be no reason to increase the revision to "4" - this should just be "3" here and below

File: audio/audacity/Portfile [hunk 2]
```diff
@@ -142,26 +138,33 @@ proc add_patchfile_ifexists {args} {
     }
 }
 
-# openssl is a dependency of curl in cmake and of python3.7
-license_noconflict  cmake python37
+# openssl is a dependency of curl in cmake and (indirectly) of git, both build dependencies
+license_noconflict  openssl
```
Comment (reneeotten) [inline audio/audacity/Portfile:142]: isn't openssl v3 used for these build dependencies? Of so, there shouldn't be a conflict anymore and the line is redundant?

File: audio/audacity/Portfile [hunk 3]
```diff
@@ -142,26 +138,33 @@ proc add_patchfile_ifexists {args} {
...
+use_parallel_build  yes
```
Comment (reneeotten) [inline audio/audacity/Portfile:143]: this is the default, so the line can/should be removed

File: audio/audacity/Portfile [hunk 4]
```diff
@@ -264,16 +267,54 @@ platform darwin {
 # the wx build system no longer exports private libraries. Carbon could be temporary because wxW
 # dropped Carbon usage somewhere around 3.1.4 but drops support for 10.9- earlier. This is not
 # required when building against port:wxWidgets-3.1 but it won't hurt either.
-patchfiles-append   ${PPREFIX}patch-cmakefiles.diff
+patchfiles-append   ${PPREFIX}patch-cmakefiles-@prefix@.diff
 
+# make export work with ffmpeg 4.4:
+patchfiles-append   patch-use-ffmpeg-4.4.diff
+
+if {${os.major} >= 17} {
```
Comment (reneeotten) [inline audio/audacity/Portfile:275]: this should include a check for "os.platform" here and below

**Review by reneeotten — COMMENTED** [review]
A few small comments after a cursory look

**Comment by RJVB** [comment]
>
>this should include a check for "os.platform" here and below

Actually it doesn't because the instructions are moot on other platforms.

Where do you see another unconditional use of `os.major`?

**Comment by RJVB** [comment]
NB: I have resolved the darwin 20+ build failure *tentatively* by disabling the SDK 10.13 workaround on all OS versions except the [10.14,11.0> range. According to upstream the hack is still required on 11.0 and up (because "To be able to raise the target we need to rewrite how the painting is done."), so someone would have to confirm this. Ideally both on Intel and ARM hw.

The commit message says this addresses "playback meters slow to respond" in zoomed mode but I haven't yet found the original bug (2400) report.

**Comment by reneeotten** [comment]
> >this should include a check for "os.platform" here and below
> 
> Actually it doesn't because the instructions are moot on other platforms.
> 
> Where do you see another unconditional use of `os.major`?

this will not work on other systems (e.g., linux) and parsing of the Portfile will likely fail. You will see that these two checks (almost) always occur together as they should.

**Comment by RJVB** [comment]
>this will not work on other systems (e.g., linux)

I have MacPorts running on Linux (#ee6096ce378, with my own patches). I'm aware that's an older "base" version but `port patch` worked just fine without the extra condition.
None of the instructions has any effect on Linux; they set irrelevant env. variables, and `patchfiles-delete` ignores (IMHO correctly) requests to delete non-matching strings.

I prepare my PRs under Linux, which includes merging my personal versions of ports with the mainstream version, which I then test as well as possible.

**Comment by reneeotten** [comment]
well, the same change is being asked in many, many other PRs so just please make the change.

**Comment by pmetzger** [comment]
@reneeotten This has been sitting quite a while. What's your opinion?

**Comment by reneeotten** [comment]
this PR is adding 20,000+ lines and I have no way or interest to vet that. So I don't know and will personally not merge this PR.

**Comment by RJVB** [comment]
I thought I had a newer PR for Audacity but I must be confused with VLC.

>this PR is adding 20,000+ lines and I have no way or interest to vet that. So I don't know and will personally not merge this PR. 

That made me look and indeed, something must have gone wrong on my end causing what seems a considerable number of irrelevant files to be included. Easy to see now with a side-bar display that I can't recall having seen before on GT but it would have been constructive to draw my attention to it before. 

My current version of the port  is from last August (so undoubtedly already obsolete...) but will still represent a big change because of the overhaul, introduction of the legacy port and because if anything the application has become even more annoying to keep updated than it already was. Feedback like above (and further above) don't motivate me at all to invest more time in trying to present the required changes in digestible morsels.

**Comment by pmetzger** [comment]
How about we close this for the moment?

**Comment by RJVB** [comment]
Fine with me!

**Pull request closed without merging**